### PR TITLE
OBPIH-2228 Add custom substitutions to substitution modal

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

File: grails-app/i18n/messages.properties [hunk 1]
```diff
@@ -2345,6 +2345,7 @@ react.default.password.placeholder=password
 react.default.button.login.label=Login
 react.default.button.delete.label=Delete
 react.default.button.addLine.label=Add line
+react.default.button.addCustomSubstitution.label=Add custom substitution
```
Comment (awalkowiak) [inline grails-app/i18n/messages.properties]: Where is this label used?

File: src/js/components/stock-movement-wizard/EditPage.jsx [hunk 2]
```diff
@@ -130,7 +130,7 @@ const FIELDS = {
           productCode: fieldValue.productCode,
           btnOpenText: `react.stockMovement.${fieldValue.substitutionStatus}.label`,
           btnOpenDefaultText: `${fieldValue.substitutionStatus}`,
-          btnOpenDisabled: fieldValue.substitutionStatus === 'NO' || fieldValue.statusCode === 'SUBSTITUTED' || showOnly,
+          btnOpenDisabled: fieldValue.statusCode === 'SUBSTITUTED' || showOnly,
           btnOpenClassName: BTN_CLASS_MAPPER[fieldValue.substitutionStatus || 'HIDDEN'],
```
Comment (awalkowiak) [inline src/js/components/stock-movement-wizard/EditPage.jsx]: If the substitutionStatus will be 'NO', then the css class "disabled" will be applied.

**Comment by awalkowiak** [comment]
@jmiranda please take a look on this feature

File: src/js/components/stock-movement-wizard/EditPage.jsx (hunk 2 above)
Comment (DominikaHatala) [reply src/js/components/stock-movement-wizard/EditPage.jsx]: It should stay grey when there is no substitution but should be clickable.

Comment (awalkowiak) [reply src/js/components/stock-movement-wizard/EditPage.jsx]: Grey comes from btn-outline-secondary not from disabled (it adds small opacity). Disabled class also affects the cursor. 
>.btn:not(:disabled):not(.disabled) {
    cursor: pointer;
}

Please remove disabled class.

Comment (DominikaHatala) [reply src/js/components/stock-movement-wizard/EditPage.jsx]: Done

File: grails-app/controllers/org/pih/warehouse/api/ProductApiController.groovy [hunk 3]
```diff
@@ -46,14 +46,32 @@ class ProductApiController extends BaseDomainApiController {
...
+        def location = Location.get(session.warehouse.id)
+        def availableItems
         if (params.name && params.name.size() < minLength) {
             render([data: []])
             return
         }
 
         String[] terms = params?.name?.split(",| ")?.findAll { it }
         def products = productService.searchProducts(terms, [])
+        if(params.availableItems) {
```
[the remark this answers is not in the document]
Comment (jmiranda) [reply grails-app/controllers/org/pih/warehouse/api/ProductApiController.groovy]: @DominikaHatala I'm going to merge the PR as-is but we need to discuss refactoring this on an upcoming Monday call. This is an expensive and not something I want to do. I'd rather get available items for a single product at a time.

**Pull request merged**